### ISSUE TITLE
share menu

### ISSUE THREAD

**Issue by burritojustice** [tangrams/tangram-play#649]
once we implement #637, add a "Share" menu item that generates a mapzen.com/tangram/view URL based off of the current scene.

**Comment by souperneon** [comment]
Share requires saving, we should have the button show up prominently for now. Button stays active and prompts sign-in+save to share

**Comment by louh** [comment]
![screenshot 2017-01-25 14 49 30](https://cloud.githubusercontent.com/assets/2553268/22306431/410416c6-e30d-11e6-9c52-24bb7276fd71.png)

**Comment by souperneon** [comment]
Can we try making the 'Share' more buttony feeling to both call it out as an action item in the menu and highlight it?

**Comment by burritojustice** [comment]
thanks -- and noticed the non-dev http://mapzen.com/tangram/view URLs are working.

Am assuming tangram/view is the most recent version of tangram-frame -- adjacently related I've made an issue for a URL shortening suggestion https://github.com/tangrams/tangram-frame/issues/40 -- basically https://mapzen.com/tangram/view/&api=22/148#15/37.7615/-122.4173

**Comment by souperneon** [comment]
Make share a menu and make the options-
- Public link (requires user to have signed in and saved the style)
- Embed code

**Comment by louh** [comment]
@burritojustice I'll add some more thoughts about URL shortening in the other issue, but I just wanted to clarify that for Tangram Play's goals, we will not be shortening URLs here.

**Comment by burritojustice** [comment]
working with folks outside Mapzen on a project in play, I realized we need a `Share code` option (that open's Play) as well as `Share map` (which opens the tangram view url)

**Comment by louh** [comment]
Why couldn't you just save -> copy and paste the URL?

**Comment by souperneon** [comment]
If I understand John correctly, he is talking about 2 different use cases of what Share could mean -
- A user who wants to share the cool map they made as a visual. They don't want to see the Play editor but the tangram view url
- A user who wants to share the code they used to create a cool map/shader, this opens the Play editor

@burritojustice is that right?

**Comment by burritojustice** [comment]
Correct -- opening the map in tangram view, vs opening Play with the yaml for the map visible.

**Comment by louh** [comment]
Right - so my point is the link for Play, with the YAML content, is right in the URL line for the page:

![screenshot 2017-02-02 17 52 05](https://cloud.githubusercontent.com/assets/2553268/22572052/55c1bf96-e970-11e6-8701-391835313cff.png)

You can copy-paste this and get a link:

https://mapzen.com/tangram/play/?scene=https%3A%2F%2Fmapzen.com%2Fapi%2Fscenes%2F7%2F26%2Fresources%2Fempty.yaml#9.9757/40.7667/-74.0994

Were you expecting a dropdown -> modal to display the same contents as the URL bar?

**Comment by souperneon** [comment]
You are right, but we should just do both. I understand it's a redundancy but in some cases it's not obvious if copy/pasting a link will successfully open it for a different user (especially if it is something that is saved in your account)

**Comment by louh** [comment]
Ok -- I see your point. The location bar URL is not guaranteed to open exactly the code you are seeing, especially if you do not save it first.

I'm going to punt this for later. This thread has been a series of moving goal posts, and keeping in mind the zero-one-infinity rule, there is not just two types of share links, there are potentially many kinds of shareable links. I will think on this so we know have a good plan for this and not just shoe-horned functionality.

**Comment by souperneon** [comment]
👍  thanks Lou

**Comment by nvkelso** [comment]
+1 for moving / duplicating some of the "camera" button functionality into the Share menu.
